flink-core/src/main/java/org/apache/flink/configuration/NettyShuffleEnvironmentOptions.java

```diff
@@ -54,6 +54,28 @@
 			.withDescription("Enable SSL support for the taskmanager data transport. This is applicable only when the" +
 				" global flag for internal SSL (" + SecurityOptions.SSL_INTERNAL_ENABLED.key() + ") is set to true");
 
+	/**
+	 * Boolean flag indicating whether the shuffle data will be compressed or not.
+	 *
+	 * <p>Note: Data is compressed per buffer (may be sliced buffer in pipeline mode) and compression can incur extra
+	 * CPU overhead so it is more effective for IO bounded scenario when data compression ratio is high.
+	 */
+	public static final ConfigOption<Boolean> DATA_COMPRESSION_ENABLED =
+		key("taskmanager.data.compression.enabled")
```

Comment:
On another note... what if someone wants to use compression just for `BlockingBoundedPartitions`? Actually that could be a sane/valid default: `LZ4` for partitions being written to disk and `NONE` for pipelined?

```
taskmanager.network.bounded-blocking-subpartition.compression: NONE/LZ4 (default LZ4?)
taskmanager.network.compression: NONE/LZ4 (defaul NONE?)
```
For example, are you sure that your motivating example for the compression won't be slowed down in some cases if compression is enabled for all types of partitions?

Reply:
I would suggest to keep `bounded-blocking-subpartition` out of the config keys.
It is a complicated internal class name, and would confuse the users. No user has an idea what a subpartition is, bounded, blocking. etc.

Something simple like `taskmanager.network.spilled` or `taskmanager.network.batchshuffle` would be easier to understand for users.

Reply:
I proposed this name because of existing config option `taskmanager.network.bounded-blocking-subpartition-type`. But yeah, I like `taskmanager.network.batchshuffle` or maybe `taskmanager.network.blocking-shuffle` more.

Still the main question remains: do we want to configure blocking/pipelined partition separately?

Reply:
Thanks for the comments. IMO, configuring separately for blocking and pipeline mode is better. I will adopt the proposal. 
Maybe in the future, we can make some of the network config options configurable per job. For example, by passing the job level configuration to the create method and overriding the default cluster configuration when initializing InputGate and ResultPartition in Task.